server/src/main/java/org/opensearch/action/search/TransportCreatePITAction.java

```diff
@@ -0,0 +1,341 @@
+            this.shardId = shardId;
+            this.keepAlive = keepAlive;
+        }
+
+        public ShardId getShardId() {
+            return shardId;
+        }
+
+        public TimeValue getKeepAlive() {
+            return keepAlive;
+        }
+
+        public CreateReaderContextRequest(StreamInput in) throws IOException {
+            super(in);
+            this.shardId = new ShardId(in);
+            this.keepAlive = in.readTimeValue();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            shardId.writeTo(out);
+            out.writeTimeValue(keepAlive);
+        }
+    }
+
+    public static class CreateReaderContextResponse extends SearchPhaseResult {
+        public CreateReaderContextResponse(ShardSearchContextId shardSearchContextId) {
+            this.contextId = shardSearchContextId;
+        }
+
+        public CreateReaderContextResponse(StreamInput in) throws IOException {
+            super(in);
+            contextId = new ShardSearchContextId(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            contextId.writeTo(out);
+        }
+    }
+
+    public static class UpdatePITContextRequest extends TransportRequest {
+        private final String pitId;
+        private final long keepAlive;
+
+        private final TimeValue createTime;
+        private final ShardSearchContextId searchContextId;
+
+        UpdatePITContextRequest(ShardSearchContextId searchContextId, String pitId, long keepAlive, TimeValue createTime) {
+            this.pitId = pitId;
+            this.searchContextId = searchContextId;
+            this.keepAlive = keepAlive;
+            this.createTime = createTime;
+        }
+
+        UpdatePITContextRequest(StreamInput in) throws IOException {
+            super(in);
+            pitId = in.readString();
+            keepAlive = in.readLong();
+            createTime = in.readTimeValue();
+            searchContextId = new ShardSearchContextId(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(pitId);
+            out.writeLong(keepAlive);
+            out.writeTimeValue(createTime);
+            searchContextId.writeTo(out);
+        }
+
+        public ShardSearchContextId getSearchContextId() {
+            return searchContextId;
+        }
+
+        public String getPitId() {
+            return pitId;
+        }
+
+        public String id() {
+            return this.getPitId();
+        }
+
+        public TimeValue getCreateTime() {
+            return createTime;
+        }
+
+        public long getKeepAlive() {
+            return keepAlive;
+        }
+    }
+
+    public static class UpdatePitContextResponse extends TransportResponse {
+        private final String pitId;
```

Comment:
why does update response have only pit id. Is pitId even necessary in update response. I would get more information if i know whether update was successful. A simple true or false could replace this entire object.